### PR TITLE
Circleci support

### PR DESCRIPTION
**Pull request by Ffloriel**
## Proposed changes

Add support of circleci badges

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Ffloriel** [comment]
@satazor @carsy

**Comment by carsy** [comment]
@Ffloriel awesome work! Will take a look at it today. Thank you :)

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+package-lock=false
```
Comment (satazor) [inline .npmrc]: Please don't do unrelated changes.

File: index.js [hunk 2]
```diff
@@ -18,7 +17,7 @@ const parsersKeys = Object.keys(parsers);
 // but this is too generic to be added here, so we playing safe and not applying it.
 function isBadgeUrl(url) {
     return ['.svg', '.png', '.jpg', '.gif', 'svg=true', 'png=true', 'badge', 'appveyor']
-    .some(extension => url.indexOf(extension) !== -1);
+        .some(extension => url.indexOf(extension) !== -1);
```
Comment (satazor) [inline index.js]: Please don't do unrelated changes. This is not respecting the project's eslint configuration. Do you have your editor setup correctly?

File: index.js [hunk 3]
```diff
@@ -34,20 +33,20 @@ module.exports = readme => {
...
-            badge = parsers[parsersKeys[i]](parsedUrl);
-        }
-
-        // TODO: Fallback to generic shields.io
-        // If that is the case we should add a flag indicating that it is a guess.
-
-        return badge;
-    })
-    .filter(badge => !!badge);
+        .map(url => url && url.split(')')[0]) // Ignore markdown syntax leftovers
```
Comment (satazor) [inline index.js]: Same as above.

File: lib/circleci.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+// circleci https://circleci.com/gh/${user}/${repo}/tree/${branch}.svg
```
Comment (satazor) [inline lib/circleci.js]: Remove `circleli` from the initial comment line

File: lib/circleci.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+// shields https://img.shields.io/circleci/project/github/${user}/${repo}/${branch}.svg
```
Comment (satazor) [inline lib/circleci.js]: Remove `shields` from the initial comment line

File: .npmrc (hunk 1 above)
Comment (Ffloriel) [reply .npmrc]: Removed it. The package-lock.json file is now generated. `package-lock.json` is not necessary for published package, that's why I added it.

File: index.js (hunk 2 above)
Comment (Ffloriel) [reply index.js]: There might be a problem with the configuration. `npm run lint` complains about the indentation if I changed it back.

File: package.json [hunk 6]
```diff
@@ -28,15 +28,15 @@
     "node": ">=4.0.0"
   },
   "devDependencies": {
-    "@carsy/eslint-config": "^0.3.0",
-    "ava": "^0.23.0",
+    "@carsy/eslint-config": "^0.4.0",
```
Comment (satazor) [inline package.json]: Perhaps it's because you have bumped the eslint config version?

**Comment by Ffloriel** [comment]
I downgraded the package dependencies to the version they were before. And revert the indentation the way it was.
I will send separate pull requests to update the dependencies and add codacy support.

**Comment by satazor** [comment]
Please rebase against master :D

**Comment by Ffloriel** [comment]
done :)

**Comment by satazor** [comment]
Looks good, just one minor comment.

File: package-lock.json [hunk 7]
```diff
@@ -1997,24 +1997,28 @@
       "dependencies": {
```
[the remark this answers is not in the document]
Comment (Ffloriel) [reply package-lock.json]: hmm, it seems that the dependencies went from `bundled: true` to `resolved: ...`.
I don't know enough about package-lock.json to know what could cause the generation to make this diff.

[the remark this answers is not in the document]
Comment (satazor) [reply package-lock.json]: Me neither. Will it doesn't matter much anyway as package-lock isn't published.

**Comment by satazor** [comment]
`npms-analyzer` has been deployed with this new feature.

It will take a few days for the whole ecosystem to be re-analyzed though :D
